src/components/Identity/createComponent.js

```diff
@@ -113,6 +113,24 @@ export default (processIdSyncs, config, logger, consent, eventManager) => {
       },
       onResponse({ response }) {
         const promises = [];
+
+        // If we were queuing requests until we received the identity cookie,
+        // and now we have the identity cookie, we can let the queued
+        // requests go out. Technically, we should always have an identity
+        // cookie at this point, but we check just to be sure.
+        if (deferredForIdentityCookie) {
+          if (hasIdentityCookie()) {
+            deferredForIdentityCookie.resolve();
+          } else {
+            // This logic assumes that the code setting the cookie is working as expected and that
+            // the cookie was missing from the response.
+            const error = new Error(`An identity was not set properly. Please verify that the org ID ${orgId} 
+            configured in Alloy matches the org ID specified in the edge configuration.`);
+            deferredForIdentityCookie.reject(error);
+            return Promise.reject(error);
+          }
+        }
```

Comment:
I moved this code up because I don't think we should create a legacy cookie if we don't have a legit identity cookie. I also changed this:

```
const noIdentityCookieErrorMsg = `An identity was not set properly. Please verify that the org ID ${orgId} 	
            configured in Alloy matches the org ID specified in the edge configuration.`;	
return deferredForIdentityCookie.reject(noIdentityCookieErrorMsg);
```

to this:

```
const noIdentityCookieError = new Error(
  `An identity was not set properly. Please verify that the org ID ${orgId} configured in Alloy matches the org ID specified in the edge configuration.`
);
deferredForIdentityCookie.reject(noIdentityCookieError);
return Promise.reject(noIdentityCookieError);
```

This is so that both the command that triggered the request fails with the error _and_ commands whose requests were queued also fail with the error. I believe this was the intention previously, but `deferredForIdentityCookie.reject` doesn't return anything, which is why the initial command wasn't failing properly.

BTW, this also exposed another bug that I logged: https://jira.corp.adobe.com/browse/CORE-42828